pkg/controller/cassandradatacenter/cassandradatacenter_controller.go

```diff
@@ -159,6 +162,59 @@ func (r *ReconcileCassandraDataCenter) Reconcile(request reconcile.Request) (rec
 	return reconcile.Result{}, nil
 }
 
+func populateUnsetFields(instance *cassandraoperatorv1alpha1.CassandraDataCenter) {
+
+	if len(instance.Spec.Cluster) == 0 {
+		instance.Spec.Cluster = "test-cluster"
```

Comment:
Perhaps we should leave just this field required. It does make sense that the user should choose a name for the cluster at the very least. Also, I feel uncomfortable about having a completely empty spec...

WDYT?

Reply:
dunno, @benbromhead do you want to have this automatically set when empty or it is requirement?